source/use-cases/sensitive-data-encryption.txt

```diff
@@ -40,9 +40,42 @@ Procedure
 Requirements
 ~~~~~~~~~~~~
 
-- Point to setup and install directions
-- Nullam imperdiet lorem vitae vulputate lacinia.
-- Donec eget velit tincidunt, gravida diam ac, efficitur lacus.
+MongoDB Server 4.2 Enterprise
+  * For installation instructions, refer to the 
+    `Enterprise Edition Installation Tutorials <https://docs.mongodb.com/manual/installation/#mongodb-enterprise-edition-installation-tutorials>`_.
+
+
+MongoDB Driver Compatible with CSFLE 
+  * For a list of drivers that support CSFLE, refer to the
+    `driver compatibility table <https://docs.mongodb.com/manual/core/security-client-side-encryption/#driver-compatibility-table>`_.
+
+
+AWS Key Management Service (KMS)
+  * AWS KMS is required for the secure deployment of CSFLE.
+
+  * Compatible drivers currently include integration with AWS KMS. Support for
+    other KMS providers may be added in the future.
+
+  * The client application will need AWS KMS permissions to the GenerateDataKey,
+    Encrypt, and Decrypt actions.
```

Comment:
I think we can focus this a bit more on the _what_ rather than _why_. Comments for each AWS KMS bullet:

- "AWS KMS is required for the secure deployment of CSFLE."
  - We can remove the first bullet since we explain the _why_ of a requirement in the guide. 

- "Compatible drivers currently include integration with AWS KMS. Support for other KMS providers may be added in the future."
  - "Compatible _MongoDB_ drivers currently support key management through AWS KMS. Support for other KMS providers may be added in the future."

- "The client application will need AWS KMS permissions to the GenerateDataKey, Encrypt, and Decrypt actions."
  - I believe these are, specifically, "IAM role permissions" so let's call out that they need an IAM user with these permissions
  - Let's move this to be the first bullet point since it's the only actual requirement here.
  - nit: thoughts on monospacing the permission names? e.g. ``` ``GenerateDataKey`` ```

Reply:
Thanks for the suggestions, will incorporate the first two.

I will omit the third bullet point and rely on the link Moe posted (https://docs.atlas.mongodb.com/security-aws-kms/) to explain the setup including permissions. This might be a hint that there should be guides that instruct both DBA/DeevOps and Developers to do an end-to-end setup (if AWS KMS is used).

It does not bother me to call these IAM role permissions, but I think at least Encrypt/Decrypt are "key policies" that are defined on specific customer master keys that are then added to an IAM role. Pretty confusing. See: https://docs.aws.amazon.com/kms/latest/developerguide/kms-api-permissions-reference.html

Reply:
On second thought, since that link is for encryption-at-rest, the references to Atlas might be confusing. Is it ok to copy and paste a couple of the bullet points from that article?

Reply:
Let me know what you think about the updates.

Reply:
These look good to me